docs/dev/design/sphinx-jquery.rst

```diff
@@ -0,0 +1,125 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following 2 steps need to be completed:
+
+#. A Sphinx theme or extension should depend on the python package ``sphinxcontrib-jquery``.
+#. In your extension's or theme's ``setup(app)``, call ``app.setup_extension("sphinxcontrib.jquery")``.
+
+In addition to this, we recommend extension and theme developers to log to the browser's ``console.error`` in case jQuery isn't found. The log message could for instance say::
+
+  if ($ == "undefined") console.error("<package-name> depends on sphinxcontrib-jquery. Please ensure that <package-name>.setup(app) is called or add 'sphinxcontrib-jquery' to your conf.py extensions setting.")
+
+
+Documentation project owners
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you are depending on a theme or extension that did not itself address the removal of jQuery from Sphinx 6, you can patch up your project like this:
+
+#. Add ``sphinxcontrib-jquery`` to your ``requirements.txt``.
+#. Add ``sphinxcontrib.jquery`` to your ``extensions`` setting in ``conf.py``.
+
+
+Calling ``app.setup_extension("sphinxcontrib.jquery")``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When a Sphinx theme or extension calls `setup_extension() <https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.setup_extension>`_, a call to ``sphinxcontrib.jquery.setup(app)`` will happen. Adding ``sphinxcontrib.jquery`` to a documentation project's ``conf.extensions`` will also call ``sphinxcontrib.jquery.setup(app)`` (at most once).
+
+In ``sphinxcontrib.jquery.setup(app)``, jQuery is added. The default behaviour is to detect the Sphinx version and include jQuery via `app.add_js_file <https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_js_file>`__ when Sphinx is from version 6 and up. jQuery is added at most once.
+
+
+Config value: ``jquery_force_enable``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When setting ``jquery_force_enable=True``, jQuery is added no matter the Sphinx version, but at most once. This is useful if you want to handle alternative conditions for adding jQuery.
+
+.. warning::
+
+  If you set ``jquery_force_enable=True``, you most likely should also add ``Sphinx>=6`` to your theme's/extension's dependencies since versions before this already bundles jQuery!
+
+
+Release
+-------
+
+All releases follow semantic versioning: ``<major>.<minor>.<patch>``
+
+* ``major``: Changes that break existing themes and extensions - we'll aim to only ever have ``1.x.y`` releases!
+* ``minor``: Changes that contain jQuery updates or substantial feature additions.
+* ``patch``: Bug fixes and added compatibility.
+
+For distributed packages with a longer lifecycle, we propose a loose pinning of the patch and minor versions to allow for future updates, which are likely to happen when jQuery receives security updates.
+Example: ``sphinxcontrib-jquery>=1.0.0,<2.0.0`` will include all compatible future releases, including jQuery upgrades.
```

Comment:
I think this section is outdated and we should probably remove it, since we won't be following. The 1.x and 2.x already happened.

Reply:
Yes, I don't know what we are following. @AA-Turner please observe the thoughts behind using a versioning scheme :arrow_up:

Reply:
Counter: https://hynek.me/articles/semver-will-not-save-you/

Pinning doesn't make much sense in the Python ecosystem as I see it, given the culture of longer deprecations. Given the intentionally limited scope there should be no breaking changes, but breaking is different from one man to the next.

A

Reply:
I don't think the blog post is what you think it is.

That aside, if you - as the maintainer - does not want to use SemVer, then the proposal cannot mention anything about it.

So I'll remove it. I disagree, but I will remove it :+1:

Reply:
I linked the post (one of many on both sides of the debate) as my view is that how maintainers choose to implement SemVer doesn't necessarily align with users' expectations of SemVer, and I concluded that it isn't worth the hassle. That was the read I got from Hyneks's post, though perhaps I did misinterpret it.

A

Reply:
I did also notice that you've already shipped two releases as "1.0.0" and "2.0.0", so the user-directed warnings of the blog post are spot on :smile: 

It's no biggy, and I think that we - as users of Python libraries - are rather used to projects that look like SemVer but aren't (yet). When I read a version string, I interpret it together with its release history and community... the longer the history, the more likely that it has a settled understanding of SemVer.

If we have a complicated understanding of version (as the proposal goes), then there is also a risk that we forget our own guidelines or someone else takes over the project and never reads them.

So I have no problem seeing this from another perspective, I'll remove this from the proposal.

Reply:
And definitely just want to repeat what I sad about this... no biggy! I'm sure the extension will do its job well without this and I thank you for the efforts :pray:

Reply:
As far as this section is concerned, I agree with @humitos and don't think we need to define versioning too strictly. It seems we're on a different path from the plan noted here.

I'm still :+1: on semver though. It at least gives users and maintainers *some* common language to describe backwards compatibility. There will be edge cases and bugs, but designing around these is always a trap.

Reply:
I'm fine by "not following semver" on some projects. However, if that's the decision made, I usually try to use a different version string that does not look like semver to avoid misunderstanding in communication. That is, for example, instead of using `x.y.z` and _not_ following semver, I'd prefer to use `2022.10.24` (calver, or something similar).

That said, I don't have a strong opinion on this and either is fine to me 👍🏼 

My 2 cents.

Reply:
To complicate things @humitos ... let's say _for some given project_ we are NOT using SemVer now, but we might use it in the future... then having a version scheme that _looks like_ SemVer but _isn't_ would be necessary :exploding_head: 

I agree that using a scheme that definitely doesn't look like SemVer is good if you decide to not use SemVer.

Reply:
I removed everything about versioning now, let's take it as it comes :+1: